.github/workflows/build-preview-environment.yml

```diff
@@ -0,0 +1,121 @@
+# Deploys a temporary environment for testing a version of the code when a pull request is created / updated with a 'deploy-pr' label
+name: Deploy PR Environment
+concurrency:
+  group: "deploy-${{ github.event.pull_request.head.ref }}"
+  cancel-in-progress: false
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types: [ labeled, synchronize ]
```

Comment:
Don't we want to have only labeled ?

Reply:
No, if someone makes some changes to the PR and want the images to be updated with the new code changes, we also want synchronize